step-templates/azure-devops-build-pipeline.json

```diff
@@ -0,0 +1,83 @@
n    }    \n}\nfunction Get-BuildPipelineStatus \n{\n    param (\n        $defaultUrl,\n        $pipelineId,\n        $runId\n    )\n    try {\n        $url = \"$defaultUrl/_apis/pipelines/$pipelineId/runs/$($runId)?api-version=6.0-preview.1\"\n        return Invoke-RestMethod -Uri $url -Method GET -Headers $azureDevOpsAuthenticationHeader\n    }\n    catch {\n        Write-Error \"An error occurred while getting the pipeline status:`n$($_.Exception.Message)\"\n        Exit 1\n    }  \n}\n$verificationPassed = @()\n$verificationPassed += Test-RequiredValues -variableToCheck (Get-Variable AzureDevOpsAccessKey)\n$verificationPassed += Test-RequiredValues -variableToCheck (Get-Variable AzureDevOpsOrganizationName)\n$verificationPassed += Test-RequiredValues -variableToCheck (Get-Variable AzureDevOpsProjectName) \n$verificationPassed += Test-RequiredValues -variableToCheck (Get-Variable AzureDevOpsPipelineName)\n$verificationPassed += Test-RequiredValues -variableToCheck (Get-Variable AzureDevOpsBranch)\n\nif ($verificationPassed -contains $false)\n{\n\tWrite-Error \"Required values missing. Please see output for further details.\"\n\tExit 1\n}\n\n$azureDevOpsAuthenticationHeader = @{Authorization = 'Basic ' + [Convert]::ToBase64String([Text.Encoding]::ASCII.GetBytes(\":$($AzureDevOpsAccessKey)\")) }\n\nWrite-Host \"Azure DevOps Organization Name: $AzureDevOpsOrganizationName\"\nWrite-Host \"Azure DevOps Project Name: $AzureDevOpsProjectName\"\nWrite-Host \"Azure DevOps Pipeline Name: $AzureDevOpsPipelineName\"\nWrite-Host \"Azure DevOps Branch: $AzureDevOpsBranch\"\nWrite-Host \"Azure DevOps Wait Until Completion: $AzureDevOpsWaitUntilCompletion\"\n\n$defaultUrl = \"https://dev.azure.com/$AzureDevOpsOrganizationName/$AzureDevOpsProjectName\"\n\n$buildPipelineId = Get-BuildPipelineId -defaultUrl $defaultUrl -pipeline $AzureDevOpsPipelineName\n\n$body = @\"\n{\n    \"resources\": {\n      \"repositories\": {\n        \"self\": {\n          \"refName\": \"refs/heads/$AzureDevOpsBranch\"\n        }\n      }\n    }\n  }\n\"@\n\n$run = Invoke-BuildPipeline -defaultUrl $defaultUrl -pipelineId $buildPipelineId -body $body\n\nWrite-Host \"The pipeline was successfully invoked, you can access the pipeline [here]($($run._links.web.href)).\"\n\nif ($run.state -ne \"completed\" -and $AzureDevOpsWaitUntilCompletion -eq $true)\n{\n    do \n    {\n        Write-Host \"Waiting for pipeline completion...\"\n        Start-Sleep 30\n        $status = Get-BuildPipelineStatus -defaultUrl $defaultUrl -pipelineId $buildPipelineId -runId $run.id\n        Write-Host \"Current Status: $($status.state)\"\n    }\n    while ($status.state -ne \"completed\") \n    if ($status.result -ne \"succeeded\")\n    {\n        Write-Error \"The Azure DevOps pipeline failed to complete succesfully\"\n        Exit 1\n    }\n}\nelse \n{\n    if ($run.result -ne \"succeeded\")\n    {\n        Write-Error \"The Azure DevOps pipeline failed to complete succesfully\"\n        Exit 1\n    }   \n}"
```

Comment:
1. **While loop time-out?**
```powershell
do {
        Write-Host "Waiting for pipeline completion..."
        Start-Sleep 30
        $status = Get-BuildPipelineStatus -defaultUrl $defaultUrl -pipelineId $buildPipelineId -runId $run.id
        Write-Host "Current Status: $($status.state)"
    }
    while ($status.state -ne "completed") 
    if ($status.result -ne "succeeded")
    {
        Write-Error "The Azure DevOps pipeline failed to complete succesfully"
        Exit 1
    }
```
This (in theory) could this wait indefinitely if ADO never returned the right result? It might be worth having a max wait time, and to fail if that's been met?

2. **succesfully** -> successfully
Check spelling on succesfully :smile: 

3.   **Write-Highlight?**
```powershell
Write-Host "The pipeline was successfully invoked, you can access the pipeline [here]($($run._links.web.href))."
```
could be:
```powershell
Write-Highlight "The pipeline was successfully invoked, you can access the pipeline [here]($($run._links.web.href))."
```